### PR TITLE
Update docs for user config deep merging

### PR DESCRIPTION
**Pull request by descent3me**
**What this PR does / why we need it**:
Update docs for https://github.com/spotify/gordon-janitor/pull/17 . The example was previously correct, but the wording didn't match the new behaviour. 

**Which issue(s) this PR fixes**

Fixes #

**Special notes for your reviewer**:
@spotify/alf 

**Release note**:

```release-note
-
```

### CODE REVIEW

File: docs/config.rst [hunk 1]
```diff
@@ -11,7 +11,7 @@ An example of a ``gordon-janitor.toml`` file:
     :language: ini
 
 
-You may choose to have a ``gordon-janitor-user.toml`` file for development. Any top-level key will override what's found in ``gordon-janitor.toml``.
+You may choose to have a ``gordon-janitor-user.toml`` file for development. All tables are deep merged into ``gordon-janitor.toml``, to limit the amount of config duplication needed. For example, you can reset ``core.debug`` without having to redeclare which plugins you'd like to run).
```
Comment (ocsw) [inline docs/config.rst]: I'm not sure it's clear what 'deep merged' means.  'Recursively merged, with user keys overriding global ones'?

**Review by ocsw — CHANGES_REQUESTED** [review]

File: docs/config.rst (hunk 1 above)
Comment (descent3me) [reply docs/config.rst]: When I was creating the initial PR I chose deep with the analogy to deep copy. Googling around, it seems like "deep merge" is more common than "recursively merge", at least with Python dictionaries. 

I have updated the wording to include "override" in the example. Does that make it more clear?

File: docs/config.rst [hunk 2]
```diff
@@ -11,7 +11,7 @@ An example of a ``gordon-janitor.toml`` file:
     :language: ini
 
 
-You may choose to have a ``gordon-janitor-user.toml`` file for development. Any top-level key will override what's found in ``gordon-janitor.toml``.
+You may choose to have a ``gordon-janitor-user.toml`` file for development. All tables are deep merged into ``gordon-janitor.toml``, to limit the amount of config duplication needed. For example, you can override ``core.debug`` without having to redeclare which plugins you'd like to run).
```
Comment (ocsw) [inline docs/config.rst]: Maybe sections instead of tables?  You also now have an extra ).  I'm ok with it otherwise.

Comment (descent3me) [reply docs/config.rst]: Good catch. Tables is the language that toml uses. https://github.com/toml-lang/toml

Comment (ocsw) [reply docs/config.rst]: I think the question is if Jane User is going to check the official docs or just copy and modify the examples.  I want to be clear in both cases.  'tables (sections)'?

Comment (descent3me) [reply docs/config.rst]: I get that, but I think it's sufficient, given the example toml below where you can see that "core" is a section/table.

Comment (ocsw) [reply docs/config.rst]: Fair enough.

**Review by ocsw — APPROVED** [review]

**Pull request merged**